lib/view_component/codemods/v3_slot_setters.rb

```diff
@@ -0,0 +1,180 @@
+# Usage (in rails console):
+#
+# Run the codemod:
+#
+#     ViewComponent::Codemods::V3SlotSetters.new.call
+#
+# If your app uses custom paths for views, you can pass them in:
+#
+#     ViewComponent::Codemods::V3SlotSetters.new(
+#       view_path: "../app/views",
+#     ).call
+
+module ViewComponent
+  module Codemods
+    class V3SlotSetters
+      TEMPLATE_LANGUAGES = %w[erb slim haml].join(",").freeze
+      RENDER_REGEX = /render[( ](?<component>\w+(?:::\w+)*)\.new[) ]+(do|\{) \|(?<arg>\w+)\b/ # standard:disable Lint/MixedRegexpCaptureTypes
+
+      Suggestion = Struct.new(:file, :line, :message)
+
+      def initialize(view_component_path: [], view_path: [])
+        Zeitwerk::Loader.eager_load_all
+
+        @view_component_path = view_component_path
+        @view_path = view_path
+      end
+
+      def call
+        puts "Using ViewComponent path: #{view_component_paths.join(", ")}"
+        puts "Using Views path: #{view_paths.join(", ")}"
+        puts "#{view_components.size} ViewComponents found"
+        puts "#{slottable_components.size} ViewComponents using Slots found"
+        puts "#{view_component_files.size} ViewComponent templates found"
+        puts "#{view_files.size} view files found"
+        process_all_files
+      end
+
+      def process_all_files
+        all_files.each do |file|
+          process_file(file)
+        end
+      end
+
+      def process_file(file)
+        @suggestions = []
+        @suggestions += scan_exact_matches(file)
+        @suggestions += scan_uncertain_matches(file)
+
+        if @suggestions.any?
+          puts
+          puts "File: #{file}"
+          @suggestions.each do |s|
+            puts "=> line #{s.line}: #{s.message}"
+          end
+        end
+      end
+
+      private
+
+      def scan_exact_matches(file)
+        [].tap do |suggestions|
+          rendered_components = []
+          content = File.read(file)
+
+          if (render_match = content.match(RENDER_REGEX))
```

Comment:
Is this assuming only a single render_match per file?

If I render multiple different components in the same file, would this catch it?